index.html

```diff
@@ -787,36 +787,154 @@ <h4>`ThingLink`</h4>
+                to OAuth1, which is deprecated, so it should not be used.
+                The relevant OAuth2 flow, the <code>code</code> flow, instead 
+                begins with a "302 (Redirection)" to an authentication
+                server, eventually resulting in credentials (bearer tokens in the case of WoT)
+                that can be used for access.
+                To enable security bootstrapping on a wide variety of devices as well
+                as on browsers, the following constraints should be observed:
+                </p>
+                <ul>
+                <li><span class="rfc2119-assertion" id="exploration-secboot-401">
+                After initial contact with an exploration service using the URL provided
+                by an introduction mechanism, the exploration service
+                MUST reply with either an HTTP "401 (Unauthorized)" response code 
+                or and HTTP "302 (Redirection)" response code (OAuth2 only) if appropriate
+                authentication information has not been provided but access can be granted
+                when it is.</span>
+                Note that if the exploration service does not want to provide access for
+                some reason, it can ignore the request or reply with another code such as 404 or 403.
+                Also, if no authentication is required, equivalent to the 
+                <code>nosec</code> scheme, then the 
+                system can reply immediately with the requested TD as if authentication information
+                were provided.  
+                Bypassing authentication however (that is, effectively using the <code>nosec</code>
+                security scheme to provide the TD) is only
+                appropriate if the TD served as a response does not contain and cannot be used to infer 
+                Personally Identifiable Information.
+                </li>
+                <li><span class="rfc2119-assertion" id="exploration-secboot-auth">
+                A 401 HTTP response from an exploration service 
+                at an API endpoint intended to serve a TD
+                MUST include a <code>WWW-Authenticate</code> header
+                and any other headers
+                describing the required authorizations using one of the following IANA-registered
+                HTTP Authentication Schemes: Basic, Bearer, or Digest.</span>
+                For details of the requirements, the IANA registry should be consulted for
+                each of the above authentication schemes.  
+                <span class="rfc2119-assertion" id="exploration-secboot-oauth2-flows">
+                If a "302 (Redirection)" response code is provided during security bootstrapping, 
+                the OAuth2 <code>code</code> flow MUST be used,
+                with access credentials eventually being represented with bearer tokens.
```

Comment:
It's true that OAuth2 does not say that 302 is the only way to do this but I was hoping to limit the complexity of the Consumer by limiting the options; that was why I added this assertion.   Leaving it completely open (as the OAuth2 spec does) will lead to interoperability problems.  But I agree with the later changes to remove 302 from the error responses, since it is not really an error.  If there are other ways to specify redirections I'd rather just include them, e.g. " redirection can be accomplished with 302 OR X".

Reply:
I replied to your comments in the PR. Basically questioning why we need interoperability here.

Regardless, IMO, the assertion isn't really mandating what you intend but instead saying that a 302 means we need to do Oauth2 code flow, assuming 302 is exclusively used for Oauth2.

I still think the assertion isn't needed, but if you insist:
```suggestion
                Security bootstrapping with OAuth2 `code` flow MUST use the "302 (Found)"
                response to redirect the client to a URL supplied in the Location header.
```